XMonad/Prompt/Unicode.hs

```diff
@@ -0,0 +1,99 @@
+{- |
+Module      :  XMonad.Prompt.Unicode
+Copyright   :  (c) 2016 Joachim Breitner
+License     :  BSD-style (see LICENSE)
+
+Maintainer  :  <mail@joachim-breitner.de>
+Stability   :  stable
+
+A prompt for searching unicode characters by name and inserting them into
+the clipboard.
+
+Requires the file @\/usr\/share\/unicode\/UnicodeData.txt@ (shipped in the package
+@unicode-data@ on Debian) and the @xsel@ tool.
+-}
+
+module XMonad.Prompt.Unicode (
+ -- * Usage
+ -- $usage
+ unicodePrompt
+ ) where
+
+import qualified Data.ByteString.Char8 as BS
+import Data.Char
+import Data.Maybe
+import Data.Ord
+import Numeric
+import System.Environment
+import System.IO
+import System.IO.Unsafe
+import System.IO.Error
+import Control.Arrow
+import Data.List
+import Text.Printf
+
+import XMonad
+import XMonad.Util.Run
+import XMonad.Prompt
+
+{- $usage
+
+You can use this module by importing it, along with
+"XMonad.Prompt", into your ~\/.xmonad\/xmonad.hs file:
+
+> import XMonad.Prompt
+> import XMonad.Prompt.Unicode
+
+and adding an appropriate keybinding, for example:
+
+>  , ((modm .|. controlMask, xK_u), unicodePrompt def)
+
+-}
+
+unicodeDataFilename :: String
+unicodeDataFilename =  "/usr/share/unicode/UnicodeData.txt"
+
+entries :: [(Char, BS.ByteString)]
+entries = unsafePerformIO $ do
+    datE <- tryIOError $ BS.readFile unicodeDataFilename
+    case datE of
+        Left e -> do
+            hPutStrLn stderr $ "Could not read file \"" ++ unicodeDataFilename ++ "\""
+            hPutStrLn stderr $ show e
+            hPutStrLn stderr $ "Do you have unicode-data installed?"
```

Comment:
I think it'd be better to add this to documentation(and add it to something like optdepends). BTW, I haven't seen any documentation that specifies all optdepends. Such docs will make package maintainers' lifes easier.